pkg/sync/kubernetes/kubernetes_sync.go

```diff
@@ -74,6 +85,7 @@ func (k *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
 				k.Logger.Debug("configuration deleted")
 			case DefaultEventTypeReady:
 				k.Logger.Debug("notifier ready")
+				k.ready = true
```

Comment:
@Kavindu-Dodan @thisthat do you guys think it's a bit weird that we will see the provider as ready after the first data is sent to the data-sync with the [initial fetch](https://github.com/open-feature/flagd/pull/418/files#diff-6b4b20bd0ff0a642f35bc03ca46986f7e27feb576ac3c459918d86f282a33642R48)?

Reply:
Initially, I put it after the informer was configured. But I follow the suggestion of this [comment](https://github.com/open-feature/flagd/pull/418#discussion_r1111760603) since the event `DefaultEventTypeReady` cleary state the provider is ready 😅

Reply:
FYI @skyerus. What do you think?

Reply:
Ya I am conflicted on this. :thinking: